From 9ace20f6110a6b6819bee5ee24e52916b626304b Mon Sep 17 00:00:00 2001
From: Christian Heimes <christian@python.org>
Date: Thu, 6 May 2021 16:30:12 +0200
Subject: [PATCH] [3.9] bpo-38820: Test with OpenSSL 3.0.0-alpha16 (GH-25942)

Also use new make target to install FIPS provider..
(cherry picked from commit e8525567dd325527e00b3c4ce7c4ce31ff3f1a8c)

Co-authored-by: Christian Heimes <christian@python.org>
diff --git a/Tools/ssl/multissltests.py b/Tools/ssl/multissltests.py
index c2b556cf2c06..51569f82b948 100755
--- a/Tools/ssl/multissltests.py
+++ b/Tools/ssl/multissltests.py
@@ -50,7 +50,7 @@ OPENSSL_RECENT_VERSIONS = [
     "1.0.2p",
     "1.1.0i",
     "1.1.1k",
-    "3.0.0-alpha15"
+    "3.0.0-alpha16"
 ]
 
 LIBRESSL_OLD_VERSIONS = [
@@ -273,12 +273,12 @@ class AbstractBuilder(object):
         log.info("Unpacking files to {}".format(self.build_dir))
         tf.extractall(self.build_dir, members)
 
-    def _build_src(self):
+    def _build_src(self, config_args=()):
         """Now build openssl"""
         log.info("Running build in {}".format(self.build_dir))
         cwd = self.build_dir
         cmd = [
-            "./config",
+            "./config", *config_args,
             "shared", "--debug",
             "--prefix={}".format(self.install_dir)
         ]
